beacon/goclient/sync_committee_contribution.go

```diff
@@ -91,3 +91,14 @@ func (gc *goClient) GetSyncCommitteeContribution(slot phase0.Slot, selectionProo
 func (gc *goClient) SubmitSignedContributionAndProof(contribution *altair.SignedContributionAndProof) error {
 	return gc.client.SubmitSyncCommitteeContributions(gc.ctx, []*altair.SignedContributionAndProof{contribution})
 }
+
+// waitOneThirdOrValidBlock waits until one-third of the slot has transpired (SECONDS_PER_SLOT / 3 seconds after the start of slot)
+func (gc *goClient) waitOneThirdOrValidBlock(slot phase0.Slot) {
```

Comment:
I don't see where does a valid block comes in this func. its only waitOneThird

Reply:
the purpose of this func to release the waiting 1/3 of the slot time on timeout or on valid block arrival

Reply:
Again I can't see how this function is waiting to the block arrival. you might use it in another place to do so, so this function name should include only what this function does.

Reply:
was referring to diff func :)
np renamed it to `waitForOneThirdSlotDuration`